xarray/core/common.py

```diff
@@ -1107,8 +1122,45 @@ def _resample(
         grouper: Resampler
         if isinstance(freq, ResampleCompatible):
             grouper = TimeResampler(
-                freq=freq, closed=closed, label=label, origin=origin, offset=offset
+                freq=freq,
+                closed=closed,
+                label=label,
+                origin=origin,
+                offset=offset,
+                boundaries=boundaries,
             )
+
+            # Apply trim logic at the resample level if needed
+            if boundaries == "trim":
+                # First, get the resampling periods to identify incomplete ones
+                from xarray.core.groupby import ResolvedGrouper
+
+                temp_grouper = ResolvedGrouper(grouper, group, self)
+                temp_encoded = temp_grouper.encoded
+
+                # Count data points in each period
+                codes = temp_encoded.codes
+                counts = np.bincount(codes.values)
+
+                if len(counts) > 0:
+                    # Find the most common count (expected points per period)
+                    unique_counts, count_frequencies = np.unique(
+                        counts, return_counts=True
+                    )
```

Comment:
This seems wrong, for example what about a time series starting or ending in february --- this will always have fewer points even if all of february is included